### DISCUSSION TITLE
Public folder in a subdirectory on staging / prod breaks Nova

### DISCUSSION THREAD

**Discussion by arkoe** [laravel/nova-issues#4267] [Q&A]
- Laravel Version: 9.11
- Nova Version: 4.0
- PHP Version: 8.1
- Database Driver & Version: MySQL
- Operating System and Version: CentOS
- Browser type and version: Any

### Description:
Whenever you access Nova in a subdirectory, for example: `https://staging.mydomain.com/myproject/admin` it redirect you to `https://staging.mydomain.com/myproject/myproject/admin/login`. Whatever the URL is after the TLD, it will just add it towards the URL. 

For example, if the URL would've been: `https://staging.mydomain.com/this/is/a/test/admin` it will redirect you to `https://staging.mydomain.com/this/is/a/test/this/is/a/test/admin/login`

After moving the project to an actual domain name such as `https://www.myproject.com` everything works as expected.

### Detailed steps to reproduce the issue on a fresh Nova installation:
2 options:
1. After a fresh installation, move the `public` folder 1 or more subdirectories deeper (change the relative paths in `server.php` and `public/index.php`)
2. Move the whole applications some layers deep.

Basically any configuration where you can't access the application in the domains root.

**Comment by crynobone** [comment]
Although there few articles guiding installation on subdirectory for Laravel this usage is not officially supported by Laravel. Convert this to community's discussion.

**Reply by arkoe** [reply]
Laravel works fine like this - never had issues with extracting the `public` folder. The redirect also happens *after* the page load - its not a server-side redirect, but a client-side one. So it probably has something to do with `Inertia.js` in Nova.

**Reply by davidhemphill** [reply]
What @crynobone is saying is that running Laravel in a subdirectory, while it might work, is considered an antipattern by Taylor himself, and as such we don't offer support for Nova configurations using this pattern. We'll be glad to change how Nova works if things change upstream.

**Reply by arkoe** [reply]
I see, thanks @davidhemphill . Would that mean that creating a subdomain using DirectAdmin for example, is also not possible for a Laravel application using Nova? Since it essentially is a subfolder within a root domain. Or am I misunderstanding?

**Reply by davidhemphill** [reply]
I'm not sure as I'm not familiar with DirectAdmin, but if the base URL for Nova is /nova (or whatever you set it to) is the first segment, it should work fine.

**Reply by brgmn** [reply]
@davidhemphill We bought Laravel Nova and thought it would run out-of-the-box in a subdirectory - e.g. domain.test/app/nova (respecting Laravels APP_URL and APP_DIR-environment variables - /app in this case). There's no technical reason why this shouldn't work. Now we have the same problem with Nova doing strange "double redirects" when trying to login. I think this is a bug/missing feature in Inertia (see https://github.com/inertiajs/inertia-laravel/pull/503) that could be easily used to support subdirectory-NOVA-installations. So my wish would be that you folks try to push that inertia-pull-request and just update Nova to the newest version (allowing us subfolder-customers to get it up running). That would be great and I could continue recommending Nova to my large enterprise customers. Thanks!

**Reply by brgmn** [reply]
Btw. A common use case is large enterprises is that they have an old application running (legacy php or old cms) and want to migrate to Laravel step by step (redirecting parts of the software to the new /app powered by Laravel). That's where this subdirectory-requirement comes from. It's not by design, but in most cases you can't just replace the whole application in one step.
